### PR TITLE
[16.0][MIG] mail_template_substitute: Migration to 16.0

### PR DESCRIPTION
**Pull request by amkarthik**
mail_template_substitute: Migration to 16.0

Depends on https://github.com/OCA/reporting-engine/pull/700

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Wrong title name.

**Comment by amkarthik** [comment]
@pedrobaeza Modified it.
Thanks

**Review by atchuthan — APPROVED** [review]

**Comment by amkarthik** [comment]
@hbrunn Can we get this merged?

**Comment by hbrunn** [comment]
@amkarthik please remove your [last commit](https://github.com/OCA/social/pull/1120/commits/93e7c8efb309b4a6875db3cec5c66fe8097ef7b7)

**Comment by amkarthik** [comment]
@hbrunn removed it.

**Review by hbrunn — APPROVED** [review]

File: mail_template_substitute/wizards/mail_compose_message.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        return result
+
+    @api.onchange("template_id")
+    def onchange_template_id_wrapper(self):
+        substitution_template = self._get_substitution_template(
+            self.composition_mode, self.template_id, [self.res_id]
+        )
+        if substitution_template:
+            self.template_id = substitution_template
+        return super().onchange_template_id_wrapper()
```
Comment (hbrunn) [inline mail_template_substitute/wizards/mail_compose_message.py:41]: @amkarthik this fails as the super function has an underscore: https://github.com/OCA/OCB/blob/16.0/addons/mail/wizard/mail_compose_message.py#L176

please submit a fix

Comment (atchuthan) [reply mail_template_substitute/wizards/mail_compose_message.py:41]: fixed at https://github.com/OCA/social/pull/1432

**Pull request merged**